### PR TITLE
Extract function from processReportsByCluster

### PR DESCRIPTION
**Pull request by JiriPapousek**
# Description

`processReportsByCluster` is long function, which will contain more functionality in the future (Service Log intergration). This is a small refactor to prepare this function for such changes.

## Type of change

- Refactor (refactoring code, removing useless files)

## Testing steps

N/A

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: differ/differ.go [hunk 1]
```diff
@@ -294,6 +294,86 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
...
+		log.Error().Err(err).Msg(invalidJSONContent)
+		return 0
+	}
+	_, offset, err := kafkaNotifier.ProduceMessage(msgBytes)
+	if err != nil {
+		log.Error().
+			Str(errorStr, err.Error()).
+			Msg("Couldn't send notification message to kafka topic.")
+		updateNotificationRecordErrorState(storage, err, cluster, report, notifiedAt, types.NotificationBackendTarget)
+		os.Exit(ExitStatusKafkaProducerError)
```
Comment (juandspy) [inline differ/differ.go:365]: I think we shouldn't use `os.Exit` here. Returning an error should be enough

**Review by tisnik — APPROVED** [review]
LGTM

**Review by juandspy — APPROVED** [review]
LGTM in general. I think that using the producer and event target as a parameter would make this function more flexible. This way we won't need to rewrite it for the service log producer

File: differ/differ.go (hunk 1 above)
Comment (tisnik) [reply differ/differ.go:365]: @juandspy interesting, we commented about it before, right? (AFAIK)

Comment (JiriPapousek) [reply differ/differ.go:365]: I think it was a different place in code, but yes, you commented on it before. Sorry, will change it.

Comment (JiriPapousek) [reply differ/differ.go:365]: > LGTM in general. I think that using the producer and event target as a parameter would make this function more flexible. This way we won't need to rewrite it for the service log producer

Unfortunatelly as it is, the Service Log message needs a little bit more things to be done - e.g. rendering the messages. There is still a lot of common code, though.

**Comment by JiriPapousek** [comment]
> LGTM in general. I think that using the producer and event target as a parameter would make this function more flexible. This way we won't need to rewrite it for the service log producer

Unfortunatelly as it is, the Service Log message needs a little bit more things to be done - e.g. rendering the message fields. There is still a lot of common code, though.

**Pull request merged**